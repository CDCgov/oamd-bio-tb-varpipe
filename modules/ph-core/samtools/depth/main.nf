process SAMTOOLS_DEPTH {
    tag "$meta1.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container 'oamd-bio-biocontainers-samtools:1.21_783c664_v1'

    input:
    tuple val(meta1), path(bam)
    tuple val(meta2), path(intervals)

    output:
    tuple val(meta1), path("*.tsv"), emit: tsv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta1.id}"
    def positions = intervals ? "-b ${intervals}" : ""
    """
    samtools \\
        depth \\
        --threads ${task.cpus-1} \\
        $args \\
        $positions \\
        -o ${prefix}.tsv \\
        $bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
