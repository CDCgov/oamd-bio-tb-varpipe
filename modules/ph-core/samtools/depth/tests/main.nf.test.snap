{
    "test-samtools-depth-intervals": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,a89e2f8e1b66356c7500599fcadb8784"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a21701d45a94582920fea7e0ba15eec"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,a89e2f8e1b66356c7500599fcadb8784"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a21701d45a94582920fea7e0ba15eec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T10:58:25.899671"
    },
    "test-samtools-depth": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a21701d45a94582920fea7e0ba15eec"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a21701d45a94582920fea7e0ba15eec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T10:58:17.502348"
    }
}
